torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -908,13 +909,21 @@ bool TensorDomain::hasBroadcast(const std::vector<IterDomain*>& td) {
       return true;
   return false;
 }
+
 bool TensorDomain::hasReduction(const std::vector<IterDomain*>& td) {
   for (auto id : td)
     if (id->isReduction())
       return true;
   return false;
 }
 
+bool TensorDomain::hasNontrivialReduction(const std::vector<IterDomain*>& td) {
+  for (auto id : td)
+    if (id->isReduction() && !id->rawExtent()->isOneInt())
```

Comment:
Do you want to add `bool IterDomain::isTrivialReduction()`? Similar to `isImplicitBroadcast()`

Reply:
Sure! Added this in the most recent version.